lightning/src/onion_message/messenger.rs

```diff
@@ -983,6 +1065,13 @@ where
 						e.get_mut().enqueue_message(onion_message);
 						log_trace!(logger, "Forwarding an onion message to peer {}", next_node_id);
 					},
+					_ if self.intercept_oms_for_offline_peers => {
+						self.enqueue_event(
```

Comment:
Note: we could avoid introducing a lock order dependency between `message_recipients` and `pending_events` by pushing the pending events at the end of the method. I didn't bother but lmk if that seems worth it.